Source/Scene/Expression.js

```diff
@@ -56,8 +56,11 @@ define([
 
     });
 
-    Expression.prototype.evaluate = function(feature) {
-        return this._runtimeAst.evaluate(feature);
+    Expression.prototype.evaluate = function(feature, result) {
```

Comment:
The style engine needs to be updated as well to call with this a scratch variable.  Also consider doing this in a separate `evaluateColor` function.